### PR TITLE
Add tests to double-check that our dummy data could be loaded

### LINKED ISSUE #931: Test dummy data
Dummy data is used to populate a new database in dev mode. Currently it goes untested. Would be good to check that all data properly inserts to database.

### PR DESCRIPTION
**Pull request by kalvdans**
closes: https://github.com/Couchers-org/couchers/issues/931
Test loading our dummy data so that issues like https://github.com/Couchers-org/couchers/pull/1248 won't happen again.

**Backend checklist**
- [x] Formatted my code by running `isort . && black .` in `app/backend`
- [x] Added tests for any new code or added a regression test if fixing a bug
- [ ] All tests pass
- [x] Run the backend locally and it works
- [x] Added migrations if there are any database changes, rebased onto `develop` if necessary for linear migration history

### CODE REVIEW

File: app/backend/src/dummy_data.py [hunk 1]
```diff
@@ -348,3 +353,8 @@ def add_dummy_communities():
 def add_dummy_data():
     add_dummy_users()
     add_dummy_communities()
+
+
+def test_add_dummy_data(db, caplog):
+    add_dummy_data()
+    assert len(caplog.records) == 0
```
Comment (aapeliv) [inline app/backend/src/dummy_data.py]: Why not put this in `tests/`?

**Review by aapeliv — COMMENTED** [review]
Thanks for doing this!

There was also an issue explicitly for this #931

File: app/backend/src/dummy_data.py (hunk 1 above)
Comment (kalvdans) [reply app/backend/src/dummy_data.py]: Good question! I somehow thought that `tests/` should only test the `couchers/` code.

(But one could equally well say that `dummy_data.py` should run without the `tests/` code.)

What do you think?

Comment (aapeliv) [reply app/backend/src/dummy_data.py]: I was just thinking of `tests/` as where the tests go :)

Or maybe just move it into `test_dummy_data.py` next to `dummy_data.py`? I just feel mixing testing code with the main code is a bit icky?

**Comment by aapeliv** [comment]
Also, it seems this test isn't being picked up by CI?

**Comment by kalvdans** [comment]
> Also, it seems this test isn't being picked up by CI?

True! I wonder why that is. We run pytest on the `src` folder, so it should search it recursively according to documentation.

Found it! It is the [python_files = *.py](https://docs.pytest.org/en/latest/reference/reference.html?highlight=python_files#confval-python_files) setting we have at work that fooled me to that pytest by default searches everything.

File: app/backend/src/dummy_data.py (hunk 1 above)
Comment (kalvdans) [reply app/backend/src/dummy_data.py]: Ok, I both made an own file and moved it to tests :) Let's see if CI picks it up.

**Review by aapeliv — APPROVED** [review]

**Pull request merged**